### ISSUE TITLE
Return inserted entities as collection

### ISSUE THREAD

**Issue by MiguelFreire** [revolver-app/vuex-orm#229]
After an insert is it possible to get the entities in form of a collection i.e an array indexed with id's?

Right now when you insert an array of data in the store:
`store.dispatch('entities/posts/create', {data: res})}).then((entities) => {})`
 you get an object of objects { Object1, Object2, ... , ObjectN} I was looking for something like Eloquent ORM does: ['id1': object1, 'id2': object2, ... , 'idN': objectN]

Is there any way to do this?

Best regards

**Comment by kiaking** [comment]
@MiguelFreire Hi! Thank you for the feedback!

Currently, there's no way to get array back. The reason for this is, because user can pass data with relationships, and then, the created records could be more than one type of entity. For example, if you pass in User with nested Post, then, the return value will be;

```js
{
  users: [ ... ],
  posts: [ ... ]
}
```

But I have an idea of implementing some new method that returns only the top level entity something like `insertOne`. Then the return value will be array just containing the top level entity (in above example, it's User).

**Comment by MiguelFreire** [comment]
Thank you for your answer @kiaking.
I think a dictionary/indexed collection approach feels more natural. 
Like this:
```javascript
const users = {
           'id1': {
                 ...userMeta,
                  posts: {
                      'pid1': post1,
                      'pid2': post2,
                   }
              }
           ...
}
```

I can try to implement that method, and perhaps an insertWith to get the collection with all relationships just like I showed above. Do you think it's possible?

**Comment by kiaking** [comment]
Yeah, that could be one solution. I think there's more we can do with the returned object, but I would like to discuss this a bit more on edge cases first.

Your proposal is nice and I think this is what people are looking for in most of the cases. However, because Vuex ORM normalizes given records, there're few challenges that we must overcome.

Let's say we insert records of users. It could contain the same user object inside nested relation.

```js
// Record of users.
{
  id: 1,
  posts: [
    {
      id: 1,
      author: { // <- This is the same user!
        id: 1
      }
    }
  ]
}
```

Now, this example is a bit unrealistic but it could very much happen if the post contains comment and comment contains the author.

Now, in this case, we wouldn't be so sure what was the actual data that has been passed at Vuex ORM at first place. Because Vuex ORM will normalize given data by normalizr, and after normalization, we lose original structure. So tracking down the original structure after data has been processed is quite hard.

One idea is to keep all relation (ids for the related records) information and then reload records once again. This might be easier since normalizr will keep such ids after normalizing the data, but this is quite hard work for more complex relationships such as MorphTo or HasManyThruough.

I think it's going to be easier, and less confusing, to keep current strucutre, but provide other way to return only top level model (in above case, the users). Because in most of the cases, what users are really looking for is the new ID value of the created record (my opinion though). In this case, we need to come up with good API name or something though...

What do you think?

**Comment by kiaking** [comment]
This is nice feature and I would like to make it possible somehow, but haven't come up with good idea yet so I'll keep this open as an enhancement issue for now 🚀

**Comment by MiguelFreire** [comment]
> This is nice feature and I would like to make it possible somehow, but haven't come up with good idea yet so I'll keep this open as an enhancement issue for now 🚀

Sorry for the late response but I've been without any time to grab this. Whenever I have time I'll try to implement this and get back to you

Cheers

**Comment by kiaking** [comment]
Thank you so much! That would be very very helpful : )
